content/build/building/variables.md

```diff
@@ -0,0 +1,547 @@
+| [BUILDKIT_HOST](#buildkit_host)                                             | String            | Specify host to use for remote builders.             |
+| [BUILDKIT_PROGRESS](#buildkit_progress)                                     | String            | Configure type of progress output.                   |
+| [BUILDKIT_TTY_LOG_LINES](#buildkit_tty_log_lines)                           | String            | Number of log lines (for active steps in tty mode).  |
+| [BUILDX_BUILDER](#buildx_builder)                                           | String            | Specify the builder instance to use.                 |
+| [BUILDX_CONFIG](#buildx_config)                                             | String            | Specify location for configuration, state, and logs. |
+| [BUILDX_EXPERIMENTAL](#buildx_experimental)                                 | Boolean           | Turn on experimental features.                       |
+| [BUILDX_GIT_CHECK_DIRTY](#buildx_git_check_dirty)                           | Boolean           | Enable dirty Git checkout detection.                 |
+| [BUILDX_GIT_INFO](#buildx_git_info)                                         | Boolean           | Remove Git information in provenance attestations.   |
+| [BUILDX_GIT_LABELS](#buildx_git_labels)                                     | String \| Boolean | Add Git provenance labels to images.                 |
+| [BUILDX_NO_DEFAULT_ATTESTATIONS](#buildx_no_default_attestations)           | Boolean           | Turn off default provenance attestations.            |
+| [BUILDX_NO_DEFAULT_LOAD](#buildx_no_default_load)                           | Boolean           | Turn off loading images to image store by default.   |
+| [EXPERIMENTAL_BUILDKIT_SOURCE_POLICY](#experimental_buildkit_source_policy) | String            | Specify a BuildKit source policy file.               |
+
+BuildKit also supports a few additional configuration parameters. Refer to
+[BuildKit built-in build args](../../engine/reference/builder.md#buildkit-built-in-build-args).
+
+You can express Boolean values for environment variables in different ways.
+For example, `true`, `1`, and `T` all evaluate to true.
+Evaluation is done using the `strconv.ParseBool` function in the Go standard library.
+See the [reference documentation](https://pkg.go.dev/strconv#ParseBool) for details.
+
+### BUILDKIT_COLORS
+
+Changes the colors of the terminal output. Set `BUILDKIT_COLORS` to a CSV string
+in the following format:
+
+```console
+$ export BUILDKIT_COLORS="run=123,20,245:error=yellow:cancel=blue:warning=white"
+```
+
+Color values can be any valid RGB hex code, or one of the
+[BuildKit predefined colors](https://github.com/moby/buildkit/blob/master/util/progress/progressui/colors.go).
+
+Setting `NO_COLOR` to anything turns off colorized output, as recommended by
+[no-color.org](https://no-color.org/).
+
+### BUILDKIT_HOST
+
+_Introduced in [Buildx v0.9.0](../release-notes.md#090)_
```

Comment:
yes we definitely should standardize this. I think we should probably have a shortcode for it like k8s does with feature_state. That would also let us have more control over rendering, and better structure the parameters (possibly even let us specify and test for old versions to be cleaned up after a while). Maybe for a follow-up?

Reply:
"+1" on having this kind of information in a more structured form.

I recall we discussed this at some point, and know we have the "API version" annotations for some options, e.g.;

<img width="662" alt="Screenshot 2024-02-20 at 11 06 40" src="https://github.com/docker/docs/assets/1804568/879b9202-2ddd-47f5-b8af-448f5bcf6ef7">


For those we (used to? still have?) a configurable "threshold", which allows use to conditionally render those badges. That way, we can _keep_ the metadata (added in version X), but can decide whether it's still relevant to highlight/show (e.g., highlighting a feature that was added 6 or 7 years ago is no longer very informative).

If we have specific criteria, we could also use that information to automatically render "new" / "what's new" badges, such as what's being added in https://github.com/docker/docs/pull/19444 (add a "new" badge because it was added for Docker Desktop 4.28, but once we are a couple of version(s) along, we can hide that).